frontend/src/stores/GameStore.ts

```diff
@@ -31,30 +36,74 @@ class GameStore {
     makeAutoObservable(this);
   }
 
-  placeTile(row: number, column: number) {
+  placeTile(row: number, column: number, fromWebsocket: boolean) {
     const tileToChange = this.boardState.find((tile) => tile.row === row && tile.column === column);
     console.log(`tile to change: `, tileToChange);
     if (tileToChange && this.tileInHand) {
       if (validateTilePlacement(row, column)) {
         tileToChange.state = TileState.TAKEN;
         tileToChange.tile = this.tileInHand;
+        if (!fromWebsocket) {
+          const websocketMessageParser = new WebsocketMessageParser();
+          const tilePlacementMessage = new TilePlacementMessage('');
+          tilePlacementMessage.id = this.tileInHand.id;
+          tilePlacementMessage.row = row;
+          tilePlacementMessage.column = column;
+          tilePlacementMessage.rotation = this.tileInHand.rotation;
+
+          websocketMessageParser.parse(tilePlacementMessage, WebSocketEvent.SEND_TILE_PLACED);
+
+          socket.emit(
+            WebSocketEvent.SEND_TILE_PLACED,
+            `${this.tileInHand.id}_${row}_${column}_${this.tileInHand.rotation}`,
+          );
+        }
+
         extendBoard(row, column);
         activateAdjacentTiles(row, column);
         manageProjects(row, column);
         console.log(rootStore.projectStore.allProjects);
 
         this.recentlyPlacedTile = this.tileInHand;
+
+        if (fromWebsocket) {
+          const indexOfTileInHand = this.drawPile.findIndex((tile) => tile.id === this.tileInHand?.id);
+          this.drawPile.splice(indexOfTileInHand, 1);
+        }
+
         this.tileInHand = undefined;
         if (this.boardState.length > 9) {
-          this.setNextPhase();
+          !fromWebsocket && this.setNextPhase();
         }
       } else {
         openInvalidMoveModal();
       }
     }
   }
 
-  setNextPhase() {
+  setTileInHandFromWebSocket(id: string, rotation: Rotation) {
+    const tiles = GameModeParser(JSONData);
```

Comment:
we expect that every tile will be stored in this.drawPile, so we can simply find tile by Id, then we can set it's rotation and place it in the hand - without creating brand new object

Reply:
done